dash/development/_r_components_generation.py

```diff
@@ -625,3 +641,104 @@ def generate_exports(
         package_imports,
         package_suggests,
     )
+
+
+def get_r_prop_types(type_object):
+    """Mapping from the PropTypes js type object to the R type"""
+
+    def shape_or_exact():
+        return 'lists containing elements {}.\n{}'.format(
+            ', '.join(
+                "'{}'".format(t) for t in list(type_object['value'].keys())
+            ),
+            'Those elements have the following types:\n{}'.format(
+                '\n'.join(
+                    create_prop_docstring(
+                        prop_name=prop_name,
+                        type_object=prop,
+                        required=prop['required'],
+                        description=prop.get('description', ''),
+                        indent_num=1
+                    ) for prop_name, prop in
+                    list(type_object['value'].items())))
+            )
+
+    return dict(
+        array=lambda: "unnamed list",
+        bool=lambda: "logical",
+        number=lambda: "numeric",
+        string=lambda: "character",
+        object=lambda: "named list",
+        any=lambda: "logical | numeric | character | "
+                    "named list | unnamed list",
+        element=lambda: "dash component",
+        node=lambda: "a list of or a singular dash "
+                     "component, string or number",
+        # React's PropTypes.oneOf
+        enum=lambda: "a value equal to: {}".format(
+            ", ".join("{}".format(str(t["value"]))
+                      for t in type_object["value"])
+        ),
+        # React's PropTypes.oneOfType
+        union=lambda: "{}".format(
+            " | ".join(
+                "{}".format(get_r_type(subType))
+                for subType in type_object["value"]
+                if get_r_type(subType) != ""
+            )
+        ),
+        # React's PropTypes.arrayOf
+        arrayOf=lambda: (
+            "list" + ((" of {}s").format(
+                get_r_type(type_object["value"]))
+                      if get_r_type(type_object["value"]) != ""
+                      else "")
+        ),
+        # React's PropTypes.objectOf
+        objectOf=lambda: (
+            "list with named elements and values of type {}"
+            ).format(
+                get_r_type(type_object["value"])
+            ),
+
+        # React's PropTypes.shape
+        shape=shape_or_exact,
+        # React's PropTypes.exact
+        exact=shape_or_exact
+    )
+
+
+def get_r_type(type_object, is_flow_type=False, indent_num=0):
```

Comment:
Please make an issue to come back to flow types. Don't do it in this PR, since we don't have any flow-typed packages of our own so we don't have an easy manual test. Come back to it after we have a test framework in place.

Reply:
Issue https://github.com/plotly/dash/issues/763 will be closed when this is resolved.